pkgs/development/python-modules/private-gpt/default.nix

```diff
@@ -0,0 +1,119 @@
+{ lib
+, buildPythonPackage
+, python
+, fetchFromGitHub
+, poetry-core
+, fastapi
+, injector
+, llama-index-core
+, llama-index-readers-file
+, huggingface-hub
+, python-multipart
+, pyyaml
+, transformers
+, uvicorn
+, watchdog
+, gradio
+, fetchurl
+, fetchpatch
+}:
+
+buildPythonPackage rec {
+  pname = "private-gpt";
+  version = "0.5.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "zylon-ai";
+    repo = "private-gpt";
+    rev = "v${version}";
+    hash = "sha256-bjydzJhOJjmbflcJbuMyNsmby7HtNPFW3MY2Tw12cHw=";
+  };
+
+  patches = [
+    # Fix a vulnerability, to be removed in the next bump version
+    # See https://github.com/zylon-ai/private-gpt/pull/1890
+    (fetchpatch {
+      url = "https://github.com/zylon-ai/private-gpt/commit/86368c61760c9cee5d977131d23ad2a3e063cbe9.patch";
+      hash = "sha256-4ysRUuNaHW4bmNzg4fn++89b430LP6AzYDoX2HplVH0=";
+    })
+  ];
```

Comment:
Would request the version bump upstream prior to merging this as for security things i am concerned about the scenario where the patch doesn't fix all issues, but appears mergable on quick look

https://github.com/zylon-ai/private-gpt/pull/1834

Reply:
Requested at https://github.com/zylon-ai/private-gpt/pull/1890#issuecomment-2102690392

Reply:
Considering https://github.com/zylon-ai/private-gpt/pull/1890#issuecomment-2102690392 as sufficient management of this problem, LGTM to merge it with this patch and if they bump the version then please update it in the nixpkgs as well.